spec/plugins/data_table/server_side.js

```diff
@@ -747,6 +747,68 @@ describe("DataTable Plugin", function() {
       expect(uri).toMatch(expectedFilterJson);
     }
 
+    it("should not duplicate filter view when there are duplicate title names", function() {
+
+      // Create a brand new table with duplicate titles
+      Backdraft.app.destroyAll();
+      app = Backdraft.app("myapp", {
+        plugins : [ "DataTable" ]
+      });
+
+      app.model("M", {});
+      app.collection("Col", {
+        model : app.Models.M,
+        url : "/somewhere"
+      });
+
+      app.view.dataTable.row("R", {
+        columns : [
+          { bulk : true },
+          { attr : "name", title : "Name", filter : { type : "string" } },
+          { attr : "cost", title : "Cost", filter : { type : "numeric" } },
+          { attr : "cost2", title : "Cost", filter : { type : "numeric" } },
+          { attr : "type", title : "Type", filter : { type : "list", options: ["Basic", "Advanced"] } },
+          { title : "Non attr column"}
+        ],
+        renderers: {
+          "Non attr column": function(node, config) {
+            node.html("Non attr column");
+          }
+        }
+      });
+      app.view.dataTable("T", {
+        rowClassName : "R",
+        serverSide : true,
+        sorting : [['Name', 'desc']],
+        filteringEnabled: true,
+        serverSideFiltering : true
+      });
+
+      app.view.dataTable.row("RSimple", {
+        columns : [
+          { attr : "cost", title : "Cost", filter : { type : "numeric" } },
+          { attr : "name", title : "Name", filter : { type : "string" } }
+        ]
+      });
+
+      app.view.dataTable("TSimple", {
+        rowClassName : "RSimple",
+        serverSide : true,
+        sorting : [['Name', 'desc']],
+        simpleParams : true
+      });
+
+      collection = new app.Collections.Col();
+
+      jasmine.Ajax.install();
+      mockResponse = new MockResponse();
+
+      table = new app.Views.T({ collection : collection });
+      table.render();
+      jasmine.Ajax.requests.mostRecent().response(mockResponse.get());
+      expect(table).toBeDefined();
```

Comment:
Yes, the test will fail when there are duplicate filters table will not be defined. On top of that the test will error out because of a duplicate filter view.